
support multiple programs to view given file

this patch introduce helper function and extend pdf program list
usually user install only popular choice of program list

order rules:
1. old known first (keep compatible characteristics with older mc versions)
2. viewer (faster then editor)
3. editor

(tested with bash and busybox sh, please cross check with ksh)

Signed-off-by: Andreas Mohr <and@gmx.li>

--- a/misc/ext.d/doc.sh.in
+++ b/misc/ext.d/doc.sh.in
@@ -10,6 +10,23 @@
 
 STAROFFICE_REGEXP='\.(sxw|sdw|stw|sxc|stc|sxi|sti|sxd|std||sxm||sxg)$'
 
+# order rules:
+# 1. old known first (keep compatible characteristics with older mc versions)
+# 2. viewer (faster then editor)
+# 3. editor
+helper_which_program() {
+    _program_list=$*
+
+    for _program in ${_program_list}; do
+        which "${_program}" >/dev/null 2>&1 || continue
+        echo "${_program}"
+        break
+    done
+    unset _program
+    unset _program_list
+    return
+}
+
 staroffice_console() {
     filename=$1;shift
     is_view=$1; shift
@@ -100,13 +117,11 @@
     pdf)
         if [ ! -n "$DISPLAY" ]; then
             pdftotext -layout -nopgbrk "${MC_EXT_FILENAME}" - | ${PAGER:-more}
-        elif see > /dev/null 2>&1; then
-            (see "${MC_EXT_FILENAME}" &)
         else
-            (xpdf "${MC_EXT_FILENAME}" &)
+            _program_caller=`helper_which_program see xpdf acroread ghostview mupdf`
+            [ "x${_program_caller}" = "x" ] || (${_program_caller} "${MC_EXT_FILENAME}" &)
+            unset _program_caller
         fi
-        #(acroread "${MC_EXT_FILENAME}" &)
-        #(ghostview "${MC_EXT_FILENAME}" &)
         ;;
     ooffice)
         if [ -n "$DISPLAY" ]; then
